File: src/kg4rd/visualization/backend/services/link_service.py
# -*- coding: utf-8 -*-
# Create Date: 2025/09/19
# Author: wangtao <wangtao.cpu@gmail.com>
# File Name: link_service.py
# Description: 链接预测

from repos.model_repo import ModelRepository
from models.link_model import LinkRequest, LinkResult
from models.result_model import Result, ResultCode

class LinkService:
    def __init__(self, model_repo: ModelRepository):
        self.model_repo = model_repo

    def link_predict(self, request: LinkRequest) -> Result[LinkResult]:
        return Result(
            code=ResultCode.QUERY_OK,
            message="Link predict success.",
            data=self.model_repo.link_predict(request)
        )
        
    def get_model_names(self) -> Result[list[str]]:
        return Result(
            code=ResultCode.QUERY_OK,
            message="Get model names success.",
            data=self.model_repo.get_model_names()
        )

File: src/kg4rd/visualization/kg4rd/src/services/link_service.ts
/**
 * 链接预测服务
 * 对应后端 link_service.py
 */

import { api } from './api'
import type { LinkRequest, LinkResult, Result } from '../types'

export class LinkService {
  private readonly baseUrl = '/link'

  async predict(request: LinkRequest): Promise<Result<LinkResult>> {
    return api.post<LinkResult>(`${this.baseUrl}/predict`, request)
  }

  async getModelNames(): Promise<Result<string[]>> {
    return api.get<string[]>(`${this.baseUrl}/model_names`)
  }
}

export const linkService = new LinkService()